### DISCUSSION TITLE
USD Selection Mode doesn't work with translated prims via mayaUsdEditAsMaya command

### DISCUSSION THREAD

**Discussion by NickWu** [Autodesk/maya-usd#2925] [Ideas]
**Describe the bug**
USD Selection Mode ( you can access it by going to Maya's main menu **Select -> USD Selection Mode**)  works fine with prims loaded with `mayaUsdProxyShape`. But once a prim is translated to Maya via `cmds.mayaUsdEditAsMaya` command, USD Selection Mode doesn't work with the translated prim anymore. 

**Steps to reproduce**
1. Load the attached "usdkindselection.usda" file with `mayaUsdProxyShape`. At this point, you can select the mesh prims individually by selecting any of them from the viewport.  
<img width="500" alt="Screenshot 2023-03-03 at 17 41 18" src="https://user-images.githubusercontent.com/935278/222868923-f8c1b23b-4c04-4d9f-8d84-a5149e91af84.png">

2. Select Maya's main menu **Select -> USD Selection Mode -> Group**
3. Now, when you select any of the mesh prims from the viewport,  their parent **box** prim will be selected instead because its `kind` value is `group`. From now on, you are not able to select individual mesh prims from the viewport. 
<img width="500" alt="Screenshot 2023-03-03 at 17 42 58" src="https://user-images.githubusercontent.com/935278/222869045-3c2cbbf9-f6d5-4b13-b036-8c93aaaeda88.png">

4. Next, with **box** prim selected, right-click in the Outliner and select "Edit As Maya Data" to translate the box prim and its children prims to become Maya native objects. 
<img width="500" alt="Screenshot 2023-03-03 at 17 49 22" src="https://user-images.githubusercontent.com/935278/222869329-f82d0589-3dd4-4fc8-8129-b325dc809784.png">

5. Now when you select any of the translated mesh prims from the viewport, you are able to select them individually. That means USD Selection Mode doesn't work with translated prims. However, USD Selection Mode should be able to tell a translated prim's kind by accessing its `kind` attribute. 
<img width="1702" alt="Screenshot 2023-03-03 at 17 50 26" src="https://user-images.githubusercontent.com/935278/222869424-2a2c66eb-019f-4e49-9a3f-1c2e1173595b.png">

**Expected behavior**
USD Selection Mode should work with USD prims as well as translated prims via `cmds.mayaUsdEditAsMaya` because they both can be accessed by UFE and a translated prim's kind value is accessible from its `kind` attribute. 

**Attachments**
usdkindselection.usda

```
#usda 1.0

def Xform "root"
{

    def Xform "house" (
        kind = "environment"
    )
    {
        def Xform "box" (
           kind = "group"
        )
        {
            def Mesh "Mesh1" (
                kind = "component"
            )
            {
                float3[] extent = [(-0.5, -0.5, -0.5), (0.5, 0.5, 0.5)]
                token faceVaryingLinearInterpolation = "all"
                int[] faceVertexCounts = [4, 4, 4, 4, 4, 4]
                int[] faceVertexIndices = [0, 1, 3, 2, 2, 3, 5, 4, 4, 5, 7, 6, 6, 7, 1, 0, 1, 7, 5, 3, 6, 0, 2, 4]
                token interpolateBoundary = "edgeOnly"
                point3f[] points = [(-0.5, -0.5, 0.5), (0.5, -0.5, 0.5), (-0.5, 0.5, 0.5), (0.5, 0.5, 0.5), (-0.5, 0.5, -0.5), (0.5, 0.5, -0.5), (-0.5, -0.5, -0.5), (0.5, -0.5, -0.5)]
                color3f[] primvars:displayColor = [(0.13320851, 0.13320851, 0.13320851)] (
                    customData = {
                        dictionary Maya = {
                            bool generated = 1
                        }
                    }
                )
                texCoord2f[] primvars:st = [(0.375, 0), (0.625, 0), (0.625, 0.25), (0.375, 0.25), (0.625, 0.5), (0.375, 0.5), (0.625, 0.75), (0.375, 0.75), (0.625, 1), (0.375, 1), (0.875, 0), (0.875, 0.25), (0.125, 0), (0.125, 0.25)] (
                    interpolation = "faceVarying"
                )
                int[] primvars:st:indices = [0, 1, 2, 3, 3, 2, 4, 5, 5, 4, 6, 7, 7, 6, 8, 9, 1, 10, 11, 2, 12, 0, 3, 13]
            }

             def Mesh "Mesh2" (
                kind = "component"
            )
            {
                double3 xformOp:translate = (2, 0, 0)
                uniform token[] xformOpOrder = ["xformOp:translate"]
                float3[] extent = [(-0.5, -0.5, -0.5), (0.5, 0.5, 0.5)]
                token faceVaryingLinearInterpolation = "all"
                int[] faceVertexCounts = [4, 4, 4, 4, 4, 4]
                int[] faceVertexIndices = [0, 1, 3, 2, 2, 3, 5, 4, 4, 5, 7, 6, 6, 7, 1, 0, 1, 7, 5, 3, 6, 0, 2, 4]
                token interpolateBoundary = "edgeOnly"
                point3f[] points = [(-0.5, -0.5, 0.5), (0.5, -0.5, 0.5), (-0.5, 0.5, 0.5), (0.5, 0.5, 0.5), (-0.5, 0.5, -0.5), (0.5, 0.5, -0.5), (-0.5, -0.5, -0.5), (0.5, -0.5, -0.5)]
                color3f[] primvars:displayColor = [(0.13320851, 0.13320851, 0.13320851)] (
                    customData = {
                        dictionary Maya = {
                            bool generated = 1
                        }
                    }
                )
                texCoord2f[] primvars:st = [(0.375, 0), (0.625, 0), (0.625, 0.25), (0.375, 0.25), (0.625, 0.5), (0.375, 0.5), (0.625, 0.75), (0.375, 0.75), (0.625, 1), (0.375, 1), (0.875, 0), (0.875, 0.25), (0.125, 0), (0.125, 0.25)] (
                    interpolation = "faceVarying"
                )
                int[] primvars:st:indices = [0, 1, 2, 3, 3, 2, 4, 5, 5, 4, 6, 7, 7, 6, 8, 9, 1, 10, 11, 2, 12, 0, 3, 13]
            }

        }
     }
   
}
```

**Specs (if applicable):**

- OS & version : CentOS 7.8
- Compiler & version : GCC 9.3.1
- Maya version: Maya-2023.2.0-11.0.PFIX & Maya 2022.4.0-24.0.PFIX
- Maya USD commit SHA: latest dev at https://github.com/Autodesk/maya-usd/commit/4bbc17f27d79e35d70536d3779086ffcf27594f7
- Pixar USD: official 22.11

**Comment by santosg87** [comment]
hi @NickWu thanks for reporting this.

I believe the workflow you have described here is simply not supported at the moment. basically due to the fact that when you are using the "edit as Maya" option, your cube is in the Maya Data model. given that some of the USD attrs. are not mapped as the same in maya, this would be something we would need to implement support for.  

I am not 100% sure on the viability of implementing this capability, tbh. I will bring it to the team :)

I am tempted to move this to a discussion thread instead to see what is the opinion of other users on the topic, is that ok with you?

**Comment by NickWu** [comment]
Thank you for your reply @santosg87. Does it mean we won't see this in development soon? No worries @santosg87.  Please feel free to move it.

**Comment by wallworm** [comment]
Hi @NickWu . This item is not on our near-term roadmap. The team will keep it in consideration as we move forward.

**Comment by NickWu** [comment]
No worries @wallworm .
